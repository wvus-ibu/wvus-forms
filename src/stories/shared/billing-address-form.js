import React, { Component } from "react";
import {
  WVUSForm,
  CheckboxControl,
  SelectControl,
  InputControl,
  validateRequired,
  validateEmail,
  validatePhone,
  validateZip,
  validateMin,
  validateMax,
  validateEmpty,
  USStateFieldOptions
} from "../../index";

class BillingInfoForm extends Component {
  constructor(props) {
    super(props);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  handleFormSubmit(e) {
    // Handle your form submission however you would like
    e.preventDefault();
    // fail safe to bail if they get around disabled button
    if (!this.props.formMethods.isFormValid()) return false;
  }

  render() {
    const { formMethods } = this.props;
    const submitDisabled = !formMethods.isFormValid();
    return (
      <form>
        <h2>Billing Info Form</h2>
        <InputControl
          labelClasses="first-name-label"
          inputClasses="first-name-input"
          fieldName="fname"
          fieldTitle="First Name"
          fieldClasses="wvus-field-50"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="lname"
          fieldTitle="Last Name"
          fieldClasses="wvus-field-50 wvus-field--end"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="address1"
          fieldTitle="Mailing Address Line 1"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="address2"
          fieldTitle="Mailing Address Line 2"
          optional={true}
          isValid={true}
          secondInteraction={false}
          formMethods={formMethods}
          validators={[]}
        />

        <InputControl
          fieldName="city"
          fieldTitle="City"
          fieldClasses="wvus-field-33"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <SelectControl
          fieldName="state"
          fieldOptions={USStateFieldOptions}
          fieldTitle="State"
          fieldClasses="wvus-field-33"
          formMethods={formMethods}
          validators={[validateRequired, validateMin(2)]}
        />

        <InputControl
          fieldName="zip"
          fieldTitle="Zip"
          fieldClasses="wvus-field-33 wvus-field--end"
          formMethods={formMethods}
          validators={[validateRequired, validateZip]}
        />
        <InputControl
          fieldName="email"
          fieldTitle="Email"
          fieldType="email"
          fieldClasses="wvus-field-newletter"
          formMethods={formMethods}
          validators={[validateRequired, validateEmail]}
        />

        <InputControl
          fieldName="phone"
          fieldTitle="Phone"
          fieldType="tel"
          fieldClasses="wvus-field-newletter"
          formMethods={formMethods}
          validators={[validateRequired, validatePhone]}
        />
        <CheckboxControl
          fieldName="optIn"
          fieldValue={true}
          fieldTitle="Yes, please add me to your email newsletter"
          labelClasses="optIn-label"
          inputClasses="optIn-input"
          fieldClasses="wvus-field-newletter-optin"
          formMethods={formMethods}
        />

        <button
          type="submit"
          id="wv-login-form-submit"
          className="btn btn-medium btn-secondary"
          disabled={submitDisabled}
          onClick={this.handleFormSubmit}
        >
          Submit
        </button>
      </form>
    );
  }
}
export { BillingInfoForm };
