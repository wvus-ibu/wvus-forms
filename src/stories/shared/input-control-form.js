import React from "react";

import {
  WVUSForm,
  InputControl,
  InputControlNeo,
  SelectControl,
  CheckboxControl
} from "../../index";
import { validateRequired } from "../../index";
import {
  formMethodsUnTouched,
  formMethodsWithSuccess,
  formMethodsWithError,
  formMethodsOptional
} from "./props-mocks";
const propsUntouched = {
  formMethods: formMethodsUnTouched
};

const propsWithSuccess = {
  formMethods: formMethodsWithSuccess
};

const propsWithError = {
  formMethods: formMethodsWithError
};

const propsOptional = {
  formMethods: formMethodsOptional
};

const propsAttrReadonlyDisabled = {
  formMethods: formMethodsUnTouched,
  attributes: {
    readOnly: true,
    disabled: true
  }
};

const propsAttrReadonlyNotDisabled = {
  formMethods: formMethodsUnTouched,
  attributes: {
    readOnly: true,
    disabled: false
  }
};
const propsAttrDisabledNotReadonly = {
  formMethods: formMethodsUnTouched,
  attributes: {
    readOnly: false,
    disabled: true
  }
};

const SimpleInputForm = props => {
  return (
    <form>
      <InputControl
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
      />
    </form>
  );
};
const SimpleInputFormHiddenMessage = props => {
  return (
    <form>
      <InputControl
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        hideMessage={true}
      />
    </form>
  );
};
const SimpleInputFormRequired = props => {
  return (
    <form>
      <InputControl
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

const SimpleInputStates = props => {
  return (
    <form>
      <InputControl
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
      />
    </form>
  );
};

const SimpleInputFormAttr = props => {
  return (
    <form>
      <InputControl
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        attributes={props.attributes}
      />
    </form>
  );
};
const SimpleInputFormClasses = props => {
  return (
    <form>
      <InputControl
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        attributes={props.attributes}
      />
    </form>
  );
};

const SimpleInputNeo = props => {
  return (
    <form>
      <InputControlNeo
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        attributes={props.attributes}
      />
    </form>
  );
};

const SimpleInputPrefixNeo = props => {
  return (
    <form>
      <InputControlNeo
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="55"
        fieldName="fname"
        fieldTitle="Price"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        attributes={props.attributes}
        renderFieldPrefix={<span class="giving-form__currency">$</span>}
      />
    </form>
  );
};

const SimpleInputNoLabelNeo = props => {
  return (
    <form>
      <InputControlNeo
        inputClasses="first-name-input"
        labelClasses="first-name-label"
        fieldPlaceholder="Enter a value"
        fieldName="fname"
        fieldTitle="Price"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        attributes={props.attributes}
        renderFieldPrefix={<span class="giving-form__currency">$</span>}
        hideLabel={true}
      />
    </form>
  );
};

export {
  SimpleInputForm,
  SimpleInputStates,
  SimpleInputFormHiddenMessage,
  SimpleInputFormRequired,
  SimpleInputFormAttr,
  SimpleInputFormClasses,
  SimpleInputNeo,
  SimpleInputPrefixNeo,
  SimpleInputNoLabelNeo,
  propsUntouched,
  propsWithError,
  propsWithSuccess,
  propsOptional,
  propsAttrReadonlyDisabled,
  propsAttrReadonlyNotDisabled,
  propsAttrDisabledNotReadonly
};
