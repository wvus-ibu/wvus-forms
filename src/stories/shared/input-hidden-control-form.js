import React from "react";

import { WVUSForm, InputHiddenControl } from "../../index";

const SimpleInputHiddenForm = props => {
  return (
    <form>
      <InputHiddenControl
        fieldName="campaign"
        fieldValue={1234567}
        fieldClasses="test-hidden-field"
        formMethods={props.formMethods}
      />
      <p>
        Hidden Field Value:{" "}
        {props.formMethods.getFieldState("campaign").value
          ? props.formMethods.getFieldState("campaign").value
          : ""}
      </p>
    </form>
  );
};

export { SimpleInputHiddenForm };
