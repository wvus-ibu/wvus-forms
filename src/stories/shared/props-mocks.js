import { action } from "@storybook/addon-actions";

const formMethodsUnTouched = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

const formMethodsWithSuccess = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "Scott Rees",
      secondInteraction: true,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => true
};

const formMethodsWithError = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: true,
      isValid: false,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => true,
  showUISuccess: () => false
};

/**
 * Optional fields should have these defaults:
 * isValid: true // allows form to be submitted without a value (since field is optional)
 * optional: true // ensures success state is not shown if field is empty and interacted with (see showUiSuccess function)
 */
const formMethodsOptional = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: true,
      isValid: true,
      optional: true,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

/**
 * Checkbox props
 */

const formMethodsChecked = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: true,
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

const formMethodsUnChecked = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: false,
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

export {
  formMethodsUnTouched,
  formMethodsWithError,
  formMethodsWithSuccess,
  formMethodsOptional,
  formMethodsChecked,
  formMethodsUnChecked
};
