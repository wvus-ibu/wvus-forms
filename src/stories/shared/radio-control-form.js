import React from "react";

import { WVUSForm, RadioControl } from "../../index";

const RadioForm = props => {
  return (
    <form>
      <RadioControl
        fieldId="payment_method_new_cc"
        fieldName="payment_method"
        fieldTitle={"New Credit Card"}
        formMethods={props.formMethods}
        fieldCheckedDefault={true}
      />
      <RadioControl
        fieldId="payment_method_existing_cc"
        fieldName="payment_method"
        fieldTitle={"Existing Credit Card"}
        formMethods={props.formMethods}
      />
      <RadioControl
        fieldId="payment_method_paypal"
        fieldName="payment_method"
        fieldTitle={"Paypal"}
        formMethods={props.formMethods}
      />
    </form>
  );
};

export { RadioForm };
