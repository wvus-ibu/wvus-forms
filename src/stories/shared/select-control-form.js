import React from "react";

import { WVUSForm, SelectControl, validateRequired } from "../../index";
// import { formMethodsChecked, formMethodsUnChecked } from "./props-mocks";
import USStateFieldOptions from "../../misc/us-states";

const SelectForm = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

const SelectFormPreselected = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldValue="WA"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        isValid={true}
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

const SelectFormNoConfig = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldOptions={props.fieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

export { SelectForm, SelectFormPreselected, SelectFormNoConfig };
