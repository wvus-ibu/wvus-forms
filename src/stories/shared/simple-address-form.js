import React from "react";

import {
  WVUSForm,
  InputControl,
  SelectControl,
  CheckboxControl
} from "../../index";
import {
  validateRequired,
  validateEmail,
  validatePhone,
  validateZip,
  validateMin,
  validateMax
} from "../../index";
import USStateFieldOptions from "../../misc/us-states";

const SimpleAddressForm = props => {
  return (
    <form>
      <h2>Simple Address Form</h2>
      <InputControl
        labelClasses="first-name-label"
        inputClasses="first-name-input"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="wvus-field-50"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="lname"
        fieldTitle="Last Name"
        fieldClasses="wvus-field- wvus-field--end"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="address1"
        fieldTitle="Mailing Address"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="city"
        fieldTitle="City"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <SelectControl
        fieldName="state"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired, validateMin(2)]}
      />
      <InputControl
        fieldName="zip"
        fieldTitle="Zip"
        fieldClasses="wvus-field-33 wvus-field--end"
        formMethods={props.formMethods}
        validators={[validateRequired, validateZip]}
      />
    </form>
  );
};
export { SimpleAddressForm };
