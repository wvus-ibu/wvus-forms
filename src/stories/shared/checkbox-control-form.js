import React from "react";

import { WVUSForm, CheckboxControl } from "../../index";
import { formMethodsChecked, formMethodsUnChecked } from "./props-mocks";

const propsChecked = {
  formMethods: formMethodsChecked
};

const propsUnchecked = {
  formMethods: formMethodsUnChecked
};

const propsDisabled = {
  formMethods: formMethodsChecked,
  attributes: {
    disabled: true
  }
};

const SimpleCheckboxForm = props => {
  return (
    <form>
      <CheckboxControl
        labelClasses="opt-in-label"
        inputClasses="opt-in-input"
        fieldName="optIn"
        fieldValue={true}
        fieldTitle="Yes, I would like to receive the email newsletter."
        fieldClasses="wvus-field-newletter-optin"
        formMethods={props.formMethods}
      />
    </form>
  );
};

const DisabledCheckboxForm = props => {
  return (
    <form>
      <fieldset disabled>
        <CheckboxControl
          fieldName="optIn"
          fieldValue={true}
          fieldTitle="Yes, I would like to receive the email newsletter."
          fieldClasses="wvus-field-newletter-optin"
          formMethods={props.formMethods}
        />
      </fieldset>
    </form>
  );
};

export {
  SimpleCheckboxForm,
  DisabledCheckboxForm,
  propsChecked,
  propsUnchecked,
  propsDisabled
};
