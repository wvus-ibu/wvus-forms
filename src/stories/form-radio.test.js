import React from "react";
import renderer from "react-test-renderer";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { mount } from "enzyme";
import sinon from "sinon";

import { RadioForm } from "./shared/radio-control-form";
import { WVUSForm, RadioControl } from "../index";

const Form = WVUSForm(RadioForm);

/**
 * SNAPSHOTS
 */
test("Snapshot: Radio Form renders correctly", () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Interactive - Enzyme tests
 */

describe("InputControl", function() {
  const expectedFieldsState = {
    payment_method: {
      value: "payment_method_new_cc",
      isValid: false,
      secondInteraction: false,
      errorMessage: "",
      validators: [],
      optional: false
    }
  };

  it("should save initial field state", () => {
    const wrapper = mount(<Form />);
    const actual = wrapper.state("fields");
    expect(actual).toEqual(expectedFieldsState);
  });

  it(" getFieldState should return fieldState", () => {
    const wrapper = mount(<Form />);

    const expected = expectedFieldsState.payment_method;
    const actual = wrapper.instance().getFieldState("payment_method");
    expect(expected).toEqual(actual);
  });

  it("should call setValueChange on Form HOC if radio button has changed", function() {
    sinon.spy(Form.prototype, "setValueChange");
    const wrapper = mount(<Form />);
    wrapper.find("input#payment_method_new_cc").simulate("change", {
      target: { name: "payment_method", id: "payment_method_existing_cc" }
    });
    expect(Form.prototype.setValueChange.calledOnce).toEqual(true);
  });

  it("should NOT call handleBlur on Form HOC if radio button is blurred", function() {
    sinon.spy(Form.prototype, "handleBlur");
    const wrapper = mount(<Form />);
    wrapper.find("input#payment_method_new_cc").simulate("blur");
    expect(Form.prototype.handleBlur.calledOnce).toEqual(false);
  });

  it("should call handleFocus if input is focus", function() {
    const handleFocus =  sinon.spy();
 
    const RadioForm = props => {
      return (
        <form>
        <RadioControl
          fieldId="payment_method_new_cc"
          fieldName="payment_method"
          fieldTitle={"New Credit Card"}
          labelClasses="payment-method-new-cc-label"
          inputClasses="payment-method-new-cc-input"
          handleFocus={handleFocus}
          formMethods={props.formMethods}
          fieldCheckedDefault={true}
        />
        <RadioControl
          fieldId="payment_method_existing_cc"
          fieldName="payment_method"
          fieldTitle={"Existing Credit Card"}
          handleFocus={handleFocus}
          formMethods={props.formMethods}
        />
        <RadioControl
          fieldId="payment_method_paypal"
          fieldName="payment_method"
          fieldTitle={"Paypal"}
          handleFocus={handleFocus}
          formMethods={props.formMethods}
        />
      </form>
      );
    };

    const Form = WVUSForm(RadioForm);
    
    const wrapper = mount(<Form />);
    wrapper.find("input#payment_method_new_cc").simulate("focus");
 
    expect(handleFocus.calledOnce).toEqual(true);
  });


  it("should be capable of having custom label and input classes", function() {
    const wrapper = mount(<Form />);
    expect(wrapper.find(".wvus-field-payment_method_new_cc label").hasClass("payment-method-new-cc-label")).toBe(true);
    expect(wrapper.find(".wvus-field-payment_method_new_cc input").hasClass("payment-method-new-cc-input")).toBe(true);
  });
});
