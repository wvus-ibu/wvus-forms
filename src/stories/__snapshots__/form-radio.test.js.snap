// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: Radio Form renders correctly 1`] = `
<div
  className=""
>
  <form>
    <div
      className="radio-group wvus-field-payment_method wvus-field-payment_method_new_cc "
    >
      <label
        className="radio-inline payment-method-new-cc-label"
      >
        <input
          checked={true}
          className="payment-method-new-cc-input"
          id="payment_method_new_cc"
          name="payment_method"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="payment_method_new_cc"
        />
        New Credit Card
      </label>
    </div>
    <div
      className="radio-group wvus-field-payment_method wvus-field-payment_method_existing_cc "
    >
      <label
        className="radio-inline "
      >
        <input
          checked={false}
          className=""
          id="payment_method_existing_cc"
          name="payment_method"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="payment_method_new_cc"
        />
        Existing Credit Card
      </label>
    </div>
    <div
      className="radio-group wvus-field-payment_method wvus-field-payment_method_paypal "
    >
      <label
        className="radio-inline "
      >
        <input
          checked={false}
          className=""
          id="payment_method_paypal"
          name="payment_method"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="payment_method_new_cc"
        />
        Paypal
      </label>
    </div>
  </form>
</div>
`;
